test/tls.c

```diff
@@ -196,9 +208,9 @@ check_tls(char *where, bool check_addr, void *tls_region)
 	check_inside_tls_region(&bss_var, tls_region);
 	check_inside_tls_region(&overaligned_bss_var, tls_region);
 
-        if (__non_tls_bss_start < __tdata_start + _tls_size() || __tdata_start + _tls_size() + 64 < __non_tls_bss_start) {
+        if (__non_tls_bss_start < __tdata_start + _tls_size() || __align_up(__tdata_start + _tls_size(), OVERALIGN_NON_TLS_BSS) < __non_tls_bss_start) {
```

Comment:
I'm not sure this change is correct. Why does the non-TLS bss area have to start _before_ the end of the TLS segment rounded up to a multiple of `OVERALIGN_NON_TLS_BSS`? The old version of the test was checking to make sure the non-TLS BSS started "close to" the end of the TLS segment (within 64 bytes). Maybe I just need a picture to understand what this is checking?

Reply:
Oh yeah that's not right, I actually want to check they do not overlap and that BSS starts shortly after.

Reply:
so maybe add a check for overlap and then make sure the non-TLS BSS starts within `OVERALIGN_NON_TLS_BSS` plus some small amount bytes (64 again? We love random values!)

Reply:
Done, hopefully CI is happy now.